types/mytime.go

```diff
@@ -173,7 +173,7 @@ func AddDate(year, month, day int64, ot gotime.Time) (nt gotime.Time) {
 }
 
 func calcTimeFromSec(to *MysqlTime, seconds, microseconds int) {
-	to.hour = seconds / 3600
+	to.hour = int16(seconds / 3600)
```

Comment:
How about checking if it's overflow and returning an error if it is, which is safer?

Reply:
`calcTimeFromSec` will be used [here](https://github.com/pingcap/tidb/blob/master/types/mytime.go#L359) and [here](https://github.com/pingcap/tidb/blob/master/types/time.go#L610). The max input value(`const secondsIn24Hour = 86400`) is promised to be less than max_int16.